### PR TITLE
Lights Sparking When Broken

### PR DESCRIPTION
**Pull request by ThatDan123**
CL: Broken lights now spark occasionally, which can start fires.

image

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/Lighting/LightSource.cs [hunk 1]
```diff
@@ -361,6 +380,33 @@ public override void Trigger(bool newState)
...
 	#endregion
 
+	#region Spark
+
+	private void TrySpark()
+	{
+		//Has to be broken and have power to spark
+		if(mState != LightMountState.Broken || powerState == PowerStates.Off) return;
+
+		//10% chance to do effect and not already doing an effect
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Lighting/LightSource.cs]:
```suggestion
		//25% chance to do effect and not already doing an effect
```

**Comment by corp-0** [comment]
sparks emit light?

**Comment by ThatDan123** [comment]
yup

**Comment by corp-0** [comment]
awesome

**Comment by PerfectTangent** [comment]
does this make the electrical storm random event cause the lights to spark upon shorting?

**Comment by ThatDan123** [comment]
probably if it breaks the lights

**Comment by PerfectTangent** [comment]
_neat_

**Comment by corp-0** [comment]
Looks good to me

**Comment by PerfectTangent** [comment]
hey, @ThatDan123, does this satisfy issue #2410?

**Comment by ThatDan123** [comment]
nah, but its close, the effect needs to be made into its own static function so can be called by any script. Wouldnt be too hard to change, but it currently doesnt fix that issue.

**Pull request merged**